mock_ros2_project/service_client.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+ROS2 Service Client Example
+
+This node demonstrates how to create a service client that calls
+the AddTwoInts service.
+"""
+
+import sys
+from typing import Optional
+
+import rclpy
+from rclpy.node import Node
+from example_interfaces.srv import AddTwoInts
+
+
+class ServiceClient(Node):
+    """Service client node that calls addition service.
+    
+    This node creates a service client that sends requests to the
+    AddTwoInts service and handles the responses.
+    """
+    
+    def __init__(self) -> None:
+        """Initialize the service client node."""
+        super().__init__('service_client')
+        
+        # Create client for the AddTwoInts service
+        self.cli = self.create_client(AddTwoInts, 'add_two_ints')
+        
+        # Wait for service to be available
+        while not self.cli.wait_for_service(timeout_sec=1.0):
+            self.get_logger().info('Service not available, waiting again...')
+        
+        self.req = AddTwoInts.Request()
+
+    def send_request(self, a: int, b: int) -> Optional[AddTwoInts.Response]:
+        """Send a request to the service.
+        
+        Args:
+            a: First integer to add
+            b: Second integer to add
+            
+        Returns:
+            Optional[AddTwoInts.Response]: Service response or None if failed
+        """
+        self.req.a = a
+        self.req.b = b
+        
+        self.get_logger().info(f'Sending request: {a} + {b}')
+        
+        try:
+            self.future = self.cli.call_async(self.req)
+            rclpy.spin_until_future_complete(self, self.future)
```

Comment:
Consider supplying the `timeout_sec` argument to `rclpy.spin_until_future_complete()` (and handling a timeout result) to avoid blocking indefinitely if the service server crashes after the initial availability check.

```suggestion
            rclpy.spin_until_future_complete(self, self.future, timeout_sec=5.0)
            if not self.future.done():
                self.get_logger().error('Service call timed out')
                return None
```